Matrix renaming + javadoc fixes

geofjamg opened the pull request:
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No


**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Methods renaming + javadoc fxes
getM -> getRowCount
getN -> getColumnCount
getValue -> get
setValue -> set
addValue -> add


**What is the current behavior?** *(You can also link to an open issue here)*
N/A


**What is the new behavior (if this is a feature change)?**
N/A


**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*
No, backward compatibility is ensured by deprecated methods


**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

murgeyseb commented on math/src/main/java/com/powsybl/math/matrix/Matrix.java:
```diff
@@ -42,46 +42,85 @@ static Matrix createFromColumn(double[] c, MatrixFactory matrixFactory) {
+        /**
+         * @deprecated Use {@link #onElement(int, int, double)} instead.
+         */
+        default void onValue(int i, int j, double value) {
```
Missing `@Deprecated` (thanks Sonar !)

geofjamg commented:
> Open question : do we keep extra tests on deprecated methods or only use the definitive one.
> It is OK for me as it is here (deprectaed methods will be removed so ...).

If i removed unit test on deprecated methods sonar complains code coverage decrease.
This is why i moved deprecated test on a single place to remove it easily when we will clean up the code.
